sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+            if t is a tuple of a symbol and a list, the system is evaluated numerically, at the given times in the list
+        t0 = 0 : number
+            the time t0 at which the state of the system is known
+
+
+        simplify=True
+            if True, the result is simplified before return.
+        do_integrals= Tue
+            if True, the function tries to evaluate the integrals in the solution. if False, it returns an
+            Integral object instead. Only valid for symbolic solutions, ignored otherwise
+
+        Examples
+        ========
+
+        >>> from sympy import Matrix, symbols, eye, zeros, ones, exp
+        >>> from sympy.physics.control.lti import StateSpaceModel
+        >>> t, y1, y2, omega = symbols('t, Y1, Y2, omega')
+        >>> ssm = StateSpaceModel(eye(2) * omega, zeros(2, 1), eye(2))
+
+            Symbolical evaluation
+            =====================
+
+        We can evaluate a StateSpaceModel, giving and input u and initial state
+        x0 and a symbol t:
+
+        >>> u = eye(1) * omega
+        >>> x0 = Matrix([y1, y2])
+        >>> ssm.evaluate(u, x0, t)
+        Matrix([
+        [Y1*exp(omega*t)],
+        [Y2*exp(omega*t)]])
+
+        One does not always have to integrate to obtain a solution. In this
+        case the do_integrals flag does nothing:
+
+        >>> ssm.evaluate(u, x0, t, do_integrals=False)
+        Matrix([
+        [Y1*exp(omega*t)],
+        [Y2*exp(omega*t)]])
+
+        But for more complicated examples evaluate retuns an Integral() object
+        when the flag is set:
+
+        >>> ssm = StateSpaceModel(Matrix([[-1, 1], [1, -1]]), ones(2, 1), eye(2))
+        >>> u = Matrix([exp(2 * t)])
+        >>> x0 = Matrix([1, 0])
+
+        >>> ssm.evaluate(u, x0, t)
+        Matrix([
+        [cosh(2*t)],
+        [sinh(2*t)]])
+
+        >>> ssm.evaluate(u, x0, t, do_integrals=False)
+        Matrix([
+        [Integral(exp(2*tau), (tau, 0, t)) + 1/2 + exp(-2*t)/2],
+        [Integral(exp(2*tau), (tau, 0, t)) + 1/2 - exp(-2*t)/2]])
+
+        >>> ssm.evaluate(u, x0, t, simplify=False)
+        Matrix([
+        [exp(2*t)/2 + exp(-2*t)/2],
+        [exp(2*t)/2 - exp(-2*t)/2]])
+
+            Numerical evaluation
+            ====================
+
+        If we give a tuple (t, t_list) instead of only the symbol, the system is
+        evaluated numericaly. The method returns a list of Matrices:
+
+        >>> ssm.evaluate(u, x0, (t, [0]))
+        [Matrix([
+        [1.0],
+        [  0]])]
+
+        >>> ssm.evaluate(u, x0, (t, [0, 0.2, 0.4, 0.6]))
+        [Matrix([
+        [1.0],
+        [  0]]), Matrix([
+        [ 1.08107237183845],
+        [0.410752325802816]]), Matrix([
+        [ 1.33743494630484],
+        [0.888105982187623]]), Matrix([
+        [1.81065556732437],
+        [1.50946135541217]])]
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
```

Comment:
I'm not a fan of this method as designed. There are two things that you need to support but they should be separated:

1. StateSpaceModel and TransferFucntion Model should have a "solve()" that produces the analytic symbolic homogenous solution OR the homogenous + nonhomegeous given a symbolic input function. These methods shoudl rely on the differential equations module in sympy.
2. You need to be able to `.evalf()`, `.subs()`, and `.xreplace()` to transform a symbolic or partial symbolic SS or TF to a partial numeric or fully numeric SS or TF. These methods should be derived from the Basic class in SymPy.